### DISCUSSION TITLE
Integration with Predbat

### DISCUSSION THREAD

**Discussion by MalcolmSpencer** [CJNE/ha-myenergi#543] [General]
I have been looking into Predbat https://github.com/springfall2008/batpred as a solution to optimise my charge and export and hopefully make some money.
Once we have the ability to control charge and force discharge through HA, this could be possible if the predbat config can be made to work with ha-myemerngi rather than the givenergy integration it was originally designed for. 
What are you clever peoples thoughts, is this going to be possible?

**Comment by plord12** [comment]
Currently I'm using specific home assistant rules (eg to maximize solar income).

As it happens I'll be describing some of this at the next U3ACommunities talk/discussion on Tuesday if interested (all free, sign up here https://u3acommunities.org/subscribe-to-computing-forum/ ) ... would be interested in your thinking at the meeting.

**Reply by MalcolmSpencer** [reply]
That sounds good, i'll look into signing up for that. I'm currently on octopus flux and have created some simple automations to dump excess power at the end of the peak period (7pm)  and also during saving sessions but I will probably move to an agile tarrif in the winter so will need something a bit more sophisticated to handle variable pricing.
